File: asterius/test/jsffi.hs
import System.Environment
import System.Process

main :: IO ()
main = do
  args <- getArgs
  callProcess "ahc-link" $
    [ "--input-hs",
      "test/jsffi/jsffi.hs",
      "--input-mjs",
      "test/jsffi/jsffi.mjs",
      "--export-function=mult_hs_int",
      "--export-function=mult_hs_double",
      "--export-function=putchar",
      "--run"
    ]
      <> args

File: asterius/test/jsffi/jsffi.mjs
import * as rts from "./rts.mjs";
import module from "./jsffi.wasm.mjs";
import jsffi from "./jsffi.req.mjs";

process.on("unhandledRejection", (err) => {
  throw err;
});

module
  .then((m) => rts.newAsteriusInstance(Object.assign(jsffi, { module: m })))
  .then(async (i) => {
    await i.exports.main();
    console.log(await i.exports.mult_hs_int(9n, 9n));
    console.log(await i.exports.mult_hs_double(9, 9));
    // await i.exports.putchar(BigInt("H".codePointAt(0)));
  });
